sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2264,6 +2265,21 @@ object GpuOverrides {
         override def convertToGpu(child: Expression): GpuExpression =
           GpuMakeDecimal(child, a.precision, a.scale, a.nullOnOverflow)
       }),
+    expr[CollectList](
+      "Collect a list of elements, now only supported by windowing.",
+      /* It should be 'fullAgg' eventually but now only support windowing,
+         so 'aggNotGroupByOrReduction' */
+      ExprChecks.aggNotGroupByOrReduction(
+        TypeSig.ARRAY.nested(TypeSig.commonCudfTypes + TypeSig.DECIMAL + TypeSig.STRUCT),
+        TypeSig.ARRAY.nested(TypeSig.all),
+        Seq(ParamCheck("input",
+          (TypeSig.commonCudfTypes + TypeSig.DECIMAL).nested() + TypeSig.STRUCT,
```

Comment:
again I think longer is clearer, but in this case because there is only one nested type I think it is okay.  So we don't support doing a collect list on a struct that also contains a struct?

Reply:
This should be supported. If it's not working, it might be a bug in `libcudf`. I'd be happy to investigate.

Reply:
Updated it, and i think it may support but I have not verified that, so do not cover it here. I am also confirming this with Mithun, waiting for his feedback.

Reply:
if yes, i will add it in a following PR

Reply:
Sounds good to me